libMBIN/Source/NMS/GameComponents/GcMissionSequenceConstructSettlementBuildingWithScanEvent.cs

```diff
@@ -1,16 +1,22 @@
+using System;
+using System.Collections.Generic;
+
 using libMBIN.NMS.Toolkit;
 using libMBIN.NMS.GameComponents;
 
 namespace libMBIN.NMS.GameComponents
 {
-    [NMS(Size = 0x228, GUID = 0xD657E0B02AB57A76, NameHash = 0x9DC327D85EF1681E)]
+    [NMS(Size = 0x2A8, GUID = 0x643BAFE66BCD82D6, NameHash = 0x9DC327D85EF1681E)]
     public class GcMissionSequenceConstructSettlementBuildingWithScanEvent : NMSTemplate
     {
         /* 0x000 */ public NMSString0x80 Message;
         /* 0x080 */ public NMSString0x80 MessageWhileBuilding;
         /* 0x100 */ public NMSString0x80 MessageWithItemsGathered;
-        /* 0x180 */ public NMSString0x20A ScanEvent;
-        /* 0x1A0 */ public float ForceCompleteSequenceAtStagePercentage;
-        /* 0x1A4 */ public NMSString0x80 DebugText;
+        /* 0x180 */ public NMSString0x80 MessageWhenDistant;
+        /* 0x200 */ public NMSString0x20A ScanEvent;
+        /* 0x220 */ public float ForceCompleteSequenceAtStagePercentage;
+        /* 0x224 */ public NMSString0x80 DebugText;
+        [NMS(Size = 0x4, Ignore = true)]
```

Comment:
This end padding is no longer required

Reply:
Curious, why TkModelResource and others but not these?

Reply:
TkModelResource and friends have a missing field at the end of the structure.  They are special cases.
I also noticed you added unneeded end padding to GcAtlasSendSubmitBase and GcNPCDebugSpawnData in a previous commit.

Reply:
Ok, I understand that TkModelResource is Size 0x84, field is 0x80 so we need to add 0x4 at the end.
Now, for example, GcMissionSequenceConstructSettlementBuildingWithScanEvent is Size 0x2A8, last field is 0x80 and ends at 0x2A4:  0x4 bytes are missing at the end, same as TkModelResource.  Please explain why it is different.  I surely miss something!

Reply:
GcMissionSequenceConstructSettlementBuildingWithScanEvent has an alignment of 8 but 0x2A4 is not a multiple of 8.
The need for end padding in this case was removed when I added code to realign the stream after the last field was read from or written to an mbin.

Reply:
We only need the padding bytes when there is space that is unaccounted for by both data and alignment.
Basically, there are almost no classes that need alignment any more other than some specific ones which already have it. So rule of thumb: Don't add alignment anywhere, and also don't remove it in places that there is alignment added.

Reply:
" has an alignment of 8 but 0x2A4 is not a multiple of 8."  That is what I did not get.  Thanks